core/Config.php

```diff
@@ -455,4 +457,25 @@ public static function setSetting($sectionName, $name, $value)
         $section[$name] = $value;
         self::getInstance()->$sectionName = $section;
     }
-}
+
+    /**
+     * Sanity check a config file by checking contents
+     *
+     * @param string $localPath
+     * @return bool
+     */
+    public function sanityCheck(string $localPath, $expectedContent): bool
+    {
+        clearstatcache(true, $localPath);
+
+        $content = @file_get_contents($localPath);
+
+        if (trim($content) !== trim($expectedContent)) {
+            Piwik::postEvent('Core.configFileSanityCheckFailed', [$localPath]);
```

Comment:
For the event it would be great to also add a little bit of text through a comment. The event will then be documented automatically on https://developer.matomo.org/api-reference/events

Reply:
@tsteur does the comment have to follow a specific format for it to be automatically documented?

Reply:
@JasonMortonNZ something similar to https://github.com/matomo-org/matomo/pull/18024/files#diff-8edb7d933e422f060b96277680eaada01d6ed1a7139e20963363b9973d625d12L415 where the comment is for the configFileChanged event. Basically, each variable in the array should be documented using `@param` and otherwise just regular comment AFAIK